From c5f222c553172139e4a0c5ab571263ca2578b5ff Mon Sep 17 00:00:00 2001
From: Pooventhiran G <quic_pooventh@quicinc.com>
Date: Fri, 6 Dec 2024 15:27:36 +0530
Subject: [PATCH] wifi: Do not clear connection owner port for MLO Reconfig

When stop-ap is sent for MLO AP reconfig link-removal, do not clear
connection owner NL port and AP SSID len as other link(s) will still be
active, and without the port, any new connection will not go through.

Fixes: 3a9b77cf6ba4 ("cfg80211: Don't reset ssid len and nlportid after a link removal.")

Change-Id: Iae05ae2b76ff6d298d8c12288b4860f82e04639b
Signed-off-by: Pooventhiran G <quic_pooventh@quicinc.com>
---
 net/wireless/ap.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/net/wireless/ap.c b/net/wireless/ap.c
index 5acbb4e04cee..e820e5421e31 100644
--- a/net/wireless/ap.c
+++ b/net/wireless/ap.c
@@ -41,9 +41,10 @@ static int ___cfg80211_stop_ap(struct cfg80211_registered_device *rdev,
 		memset(&wdev->links[link_id].ap.chandef, 0,
 		       sizeof(wdev->links[link_id].ap.chandef));
 
-		if (hweight16(wdev->valid_links) <= 1) {
-			/* Clear this only when there is one or lesser valid
-			 * link, otherwise consider that some link is present
+		if (!params.reconfig) {
+			/* Clear this only when the stop is NOT received for
+			 * MLO Reconfig link removal as other link(s) will
+			 * still be active.
 			 */
 			wdev->conn_owner_nlportid = 0;
 			wdev->u.ap.ssid_len = 0;
-- 
2.34.1

